ggml/src/ggml-cuda/ggml-cuda.cu

```diff
@@ -123,8 +123,15 @@ static ggml_cuda_device_info ggml_cuda_init() {
 #ifdef __HIP_PLATFORM_AMD__
     // Workaround for a rocBLAS bug when using multiple graphics cards:
     // https://github.com/ROCmSoftwarePlatform/rocBLAS/issues/1346
-    rocblas_initialize();
-    CUDA_CHECK(cudaDeviceSynchronize());
+    {
+        char version_string[64];
+        version_string[0] = '\0';
+        const rocblas_status status = rocblas_get_version_string(version_string, sizeof(version_string));
+        if (status != rocblas_status_success || version_string[0] < '4') {
```

Comment:
i don't like this too much as this will ofc fail if rocblas changes its version to 10.0.0 or whatever. I think we should make a bit more effort to parse this properly. Looking at rocblas (https://github.com/ROCm/rocBLAS/blob/59825a7367a24eed4e7e8a483820592089eaf17e/library/src/buildinfo.cpp#L29) it seams we would be on the safe side to use string_split<int> here (https://github.com/ggerganov/llama.cpp/blob/df984e014714cba4c99ef894b20b51cbcef31b16/common/common.h#L439)

however currently common.h is not used outside of the clients/examples and contains code that makes no sense in the backend.
@ggerganov maybe you can weigh in on if its ok to use this header here or if we should move function somewhere else.